### PR TITLE
Add memo, time bounds, fee and source accounts to transactions

### LINKED ISSUE #58: Add memo to transaction parsing and schema

### PR DESCRIPTION
**Pull request by charlie-wasp**
Resolves #58

### CODE REVIEW

File: src/model/transaction.ts [hunk 1]
```diff
@@ -1,4 +1,13 @@
 import stellar from "stellar-base";
+import { publicKeyFromBuffer } from "../util/xdr";
+
+type MemoValue = string | number;
+enum MemoType {
```
Comment (gzigzigzeo) [inline src/model/transaction.ts]: export?

File: src/model/transaction.ts [hunk 2]
```diff
@@ -8,6 +17,10 @@ export class Transaction {
   public result: string;
   public meta: string;
   public feeMeta: string;
+  public memo: { value: MemoValue; type: MemoType } | null = null;
```
Comment (gzigzigzeo) [inline src/model/transaction.ts]: I would define export type outside this class

File: src/model/transaction.ts [hunk 3]
```diff
@@ -8,6 +17,10 @@ export class Transaction {
...
+  public feeAmount: string;
+  public sourceAccount: string;
+  public timeBounds: [number, number] | null = null;
```
Comment (gzigzigzeo) [inline src/model/transaction.ts]: null or empty array? Not sure on it

**Comment by charlie-wasp** [comment]
> Tests? ;-)
Touché 🙂

File: src/model/transaction.ts (hunk 3 above)
Comment (charlie-wasp) [reply src/model/transaction.ts]: Well, when no time bounds are set, in XDR we have null, not the empty array, so now it's kind of mimic that. But of course, we can do it any way.

File: src/model/transaction.ts (hunk 2 above)
Comment (charlie-wasp) [reply src/model/transaction.ts]: Good catch 👍

File: src/model/transaction.ts (hunk 3 above)
Comment (gzigzigzeo) [reply src/model/transaction.ts]: Got it, no, let's keep it the same way as xdr do

**Comment by charlie-wasp** [comment]
@gzigzigzeo tests are added, memo type is extracted

**Pull request merged**